cmd/lncli/cmd_pay.go

```diff
@@ -127,34 +133,31 @@ var sendPaymentCommand = cli.Command{
 }
 
 // retrieveFeeLimit retrieves the fee limit based on the different fee limit
-// flags passed.
-func retrieveFeeLimit(ctx *cli.Context) (*lnrpc.FeeLimit, error) {
+// flags passed. It always returns a value and doesn't rely on lnd applying a
+// default.
+func retrieveFeeLimit(ctx *cli.Context, amt int64) (int64, error) {
 	switch {
+
 	case ctx.IsSet("fee_limit") && ctx.IsSet("fee_limit_percent"):
-		return nil, fmt.Errorf("either fee_limit or fee_limit_percent " +
+		return 0, fmt.Errorf("either fee_limit or fee_limit_percent " +
 			"can be set, but not both")
+
 	case ctx.IsSet("fee_limit"):
-		return &lnrpc.FeeLimit{
-			Limit: &lnrpc.FeeLimit_Fixed{
-				Fixed: ctx.Int64("fee_limit"),
-			},
-		}, nil
+		return ctx.Int64("fee_limit"), nil
+
 	case ctx.IsSet("fee_limit_percent"):
-		return &lnrpc.FeeLimit{
-			Limit: &lnrpc.FeeLimit_Percent{
-				Percent: ctx.Int64("fee_limit_percent"),
-			},
-		}, nil
+		return amt * ctx.Int64("fee_limit_percent") / 100, nil
```

Comment:
no strong preference really, but do we want to round up here, e.g. `(amt * fee_limit_percent + 99)/100`? could help with very small payments

Reply:
done